### DISCUSSION TITLE
How can I run test cases against testnet

### DISCUSSION THREAD

**Discussion by shivam2320** [aptos-labs/aptos-developer-discussions#149] [Questions] [answered]
How can I run test cases against testnet

**Comment by gregnazario** [comment]
Which kind of test cases?

Move?  TypeScript?

**Comment by shivam2320** [comment]
Move as of now
But would love to know for ts aswell

**Reply by kshitijc1506** [reply, marked as answer]
Have a look at this [discussion](https://github.com/aptos-labs/aptos-developer-discussions/discussions/144). You can read the first comment I wrote to understand how unit tests can be implemented in Move. It also includes the CLI command to run the tests and links to relevant documentation. 

Since you want to use testnet, make sure you have run the following command at the root of your directory:
`aptos init --network testnet`
Set up your `config.yaml` how you would like to. 

You can deploy the move package and also run tests using the TypeScript SDK. Here are the [docs](https://aptos.dev/sdks/ts-sdk/testing/) for that.

**Comment by shivam2320** [comment]
I have already setup profile on testnet, so if I run test cases using `aptos move test` will those run against testnet?

And if I setup testing using ts SDK then those transactions will get confirmed on testnet right?

**Reply by 0xmaayan** [reply]
By default the SDK tests run against a local network, however you can configure it to run against testnet by creating an `.env` file and set `APTOS_NETWORK=testnet`. Then, looking at this [code](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/e2e/helper.ts#L18), test would run against testnet
